trace_decoder/src/compact/compact_to_partial_trie.rs

```diff
@@ -8,183 +11,261 @@ use mpt_trie::{
 };
 
 use super::compact_prestate_processing::{
-    AccountNodeCode, AccountNodeData, CompactParsingResult, LeafNodeData, NodeEntry, WitnessEntry,
+    AccountNodeCode, AccountNodeData, CompactParsingError, CompactParsingResult, LeafNodeData,
+    NodeEntry, WitnessEntry,
 };
 use crate::{
-    types::{CodeHash, HashedAccountAddr, TrieRootHash, EMPTY_CODE_HASH, EMPTY_TRIE_HASH},
+    decoding::TrieType,
+    types::{
+        CodeHash, HashedAccountAddr, HashedAccountAddrNibbles, TrieRootHash, EMPTY_CODE_HASH,
+        EMPTY_TRIE_HASH,
+    },
     utils::{h_addr_nibs_to_h256, hash},
 };
 
-#[derive(Debug, Default)]
-pub(super) struct CompactToPartialTrieExtractionOutput {
-    pub(super) trie: HashedPartialTrie,
+pub trait CompactToPartialTrieExtractionOutput {
```

Comment:
Does this trait need to be `pub`?

Reply:
Yeah I think `super` is sufficient here, good catch.